client/jetpack-connect/controller.js

```diff
@@ -141,57 +154,72 @@ export function persistMobileAppFlow( context, next ) {
 }
 
 export function setMasterbar( context, next ) {
+	debug( 'controller: setMasterbar', context.params );
 	if ( config.isEnabled( 'jetpack/connect/mobile-app-flow' ) ) {
 		const masterbarToggle = retrieveMobileRedirect() ? hideMasterbar() : showMasterbar();
 		context.store.dispatch( masterbarToggle );
 	}
 	next();
 }
 
-export function connect( context, next ) {
-	const { path, pathname, params, query } = context;
-	const { type = false, interval } = params;
-	const analyticsPageTitle = get( type, analyticsPageTitleByType, 'Jetpack Connect' );
-	const planSlug = getPlanSlugFromFlowType( type, interval );
-
-	// Not clearing the plan here, because other flows can set the cookie before arriving here.
-	planSlug && storePlan( planSlug );
-	recordPageView( pathname, analyticsPageTitle );
+export function loginBeforeJetpackSearch( context, next ) {
+	debug( 'controller: loginBeforeJetpackSearch', context.params );
+	const { params, path } = context;
+	const { type } = params;
+	const isLoggedOut = ! getCurrentUserId( context.store.getState() );
```

Comment:
The flow for Jetpack Search is a bit different than the rest. It asks users to log in at the start of the flow.